### ISSUE TITLE
[Bug]: Realtime channel is not working properly

### ISSUE THREAD

**Issue by rafiul587** [supabase-community/supabase-kt#250]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/wiki/Troubleshooting) page for similar problems
- [X] I enabled [logging](https://github.com/supabase-community/supabase-kt/wiki/Logging) and checked the logs

### Version(s)

1.2.0-alpha-2

### Target(s)

Android

### What happened? (include your code)

When I go to a fragment create a channel and join, it works fine. But then I press back, onDestroyView is called and I leave the channel. Next time I go back to that fragment again, realtime doesn't work. 

Here is the repo link I made as a sample to test the behavior.
https://github.com/rafiul587/Supabase-Realtime-Sample

### Steps To Reproduce (optional)

1. Go to the message page and type  a message and send, the message will show in the textview below.
2. Navigate back to the previous page.
3. Go to the message page again and send a message, this time nothing shows in the textview.

### Relevant log output (optional)

```shell
Here is the discussion about the issue in detail.
https://discord.com/channels/839993398554656828/1136730903733862522
```

**Comment by jan-tennert** [comment]
The block inside the coroutine:

https://github.com/rafiul587/Supabase-Realtime-Sample/blob/89ea3b100ac17ad3da2104d29a8471512c7ebd87/app/src/main/java/com/example/supabaserealtime/MessageFragment.kt#L81-L83

never gets executed, resulting in the client not leaving the channel (causing the client receiving the "phx_close" event, when trying to join again)

**Comment by rafiul587** [comment]
OMG, I wasted two days on this and haven't even checked for once if that leave function is getting executed or not. In my original project, I used it in the onCleared() method of viewModel. 

So I know this is not a library issue but I would like to know how can I leave safely when the fragment gets destroyed since that is a suspend function? Should I move the channel create and leave function to the top level like activity or application?
